### PR TITLE
RHDEVDOCS-1278: Providing a custom IDE based on Eclipse Theia

### PR DESCRIPTION
**Pull request by azatsarynnyy**
Signed-off-by: Artem Zatsarynnyi <azatsary@redhat.com>

> Read our [Contribution guide](https://github.com/eclipse/che-docs/blob/master/CONTRIBUTING.adoc) before submitting a PR.

### What does this PR do?
Adds a description of the main steps for adding a Theia-based IDE as an editor.

### What issues does this PR fix or reference?
https://issues.redhat.com/browse/RHDEVDOCS-1278

### PR Checklist

As the author of this Pull Request I made sure that:

- [ ] `vale` has been run successfully against the PR branch
- [ ] Link checker has been run successfully against the PR branch
- [ ] Documentation describes a scenario that is already covered by QE tests, otherwise an issue has been created and acknowledged by Che QE team
- [ ] Changed article references are updated where they are used (or a redirect has been set up on the docs side):
    - [ ] Dashboard [branding.json](https://github.com/eclipse/che-dashboard/blob/master/src/components/branding/branding.json)
    - [ ] Chectl [constants.ts](https://github.com/che-incubator/chectl/blob/master/src/constants.ts)

### CODE REVIEW

File: modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc [hunk 1]
```diff
@@ -0,0 +1,62 @@
+[id="con_support-for-theia-based-ides_{context}"]
+= Theia-based IDEs
+
+This section describes how to provide a custom IDE, based on Eclipse Theia framework.
+
+To use a Theia-based IDE in {prod} as an editor, you need to prepare three main components:
+
+* docker image with your IDE
+* Che editor descriptor file - `meta.yaml`
```
Comment (ericwill) [inline modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc]: `
```suggestion
* the Che editor descriptor file - `meta.yaml`
```

File: modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc [hunk 2]
```diff
@@ -0,0 +1,62 @@
+[id="con_support-for-theia-based-ides_{context}"]
+= Theia-based IDEs
+
+This section describes how to provide a custom IDE, based on Eclipse Theia framework.
+
+To use a Theia-based IDE in {prod} as an editor, you need to prepare three main components:
+
+* docker image with your IDE
```
Comment (ericwill) [inline modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc]:
```suggestion
* a docker image containing your IDE
```

File: modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+* Che editor descriptor file - `meta.yaml`
+* replace an editor in a Devfile
```
Comment (ericwill) [inline modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc]: I don't think you need this bullet point if it's described in the `Procedure` section below

File: modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    containers:
+      - name: theia-ide
+        image: "<your-ide-image>"
+        mountSources: true
+        ports:
+          - exposedPort: 3100
+        memoryLimit: "512M"
+----
++
+`targetPort` and `exposedPort` must be the same as your Theia-based IDE is running on, inside a container.
```
Comment (ericwill) [inline modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc]:
```suggestion
`targetPort` and `exposedPort` must be the same as the Theia-based IDE running inside the container.
```

File: modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+Replace `<your-ide-image>` with the name of your IDE image.
```
Comment (ericwill) [inline modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc]:
```suggestion
Replace `<your-ide-image>` with the name of the IDE image.
```

File: modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+Your `meta.yaml` file should accessible by http(s) link.
```
Comment (ericwill) [inline modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc]:
```suggestion
The `meta.yaml` file should be publicly accessible through an HTTP(S) link.
```

File: modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+----
+apiVersion: 1.0.0
+metadata:
+  name: che-theia-based-ide
+components:
+  - type: cheEditor
+    reference: '<meta.yaml URL>'
+----
++
+`<meta.yaml URL>` should point to the `meta.yaml` file described on a previous step.
```
Comment (ericwill) [inline modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc]:
```suggestion
`<meta.yaml URL>` should point to the publicly hosted `meta.yaml` file described in the previous step.
```

File: modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+
+To use a Theia-based IDE in {prod} as an editor, you need to prepare three main components:
+
+* docker image with your IDE
+* Che editor descriptor file - `meta.yaml`
+* replace an editor in a Devfile
+
+.Procedure
+
+. Describe your IDE with an editor descriptor - `meta.yaml` file:
```
Comment (ericwill) [inline modules/end-user-guide/partials/con_support-for-theia-based-ides.adoc]:
```suggestion
. Describe the IDE with an editor descriptor - `meta.yaml` file:
```

**Review by ericwill — COMMENTED** [review]
Content wise it looks good, just left a few comments for style/clarity. I'm also not a docs guy so feel free to ignore some of them if they don't conform to the style guide :+1:

**Comment by azatsarynnyy** [comment]
Thanks @ericwill for looking at it!

**Review by yhontyk — APPROVED** [review]

**Comment by themr0c** [comment]
What is the target milestone?

**Comment by azatsarynnyy** [comment]
@themr0c Che 7.28

**Comment by azatsarynnyy** [comment]
@ericwill I've addressed all your comments. Can you approve it?

**Review by ericwill — APPROVED** [review]

**Comment by ericwill** [comment]
> @ericwill I've addressed all your comments. Can you approve it?

Done :+1:

**Pull request merged**